### ISSUE TITLE
Publish packages on a repository for logstash-forwarder

### ISSUE THREAD

**Issue by jordansissel** [elasticsearch/logstash-forwarder#385]
This task requires the next logstash-forwarder release to occur. Will mark this closed once that it is done.

**Comment by sstarcher** [comment]
+1

**Comment by ernetas** [comment]
Guys, please, don't +1 this, like @jordansissel [asked on another issue](https://github.com/elasticsearch/logstash-forwarder/issues/370#issuecomment-75478481). There is a "Subscribe" button on the right and there is no need to notify everyone that you +1ed this issue...

**Comment by jordansissel** [comment]
For future reference, I generally don't mind +1's but in this specific case I'm being hammered with complaints about a problem I have publically demonstrated that I am both aware of and working on. So, just for this task, no +1's please. For anything else, happy to have them for now :)

**Comment by jordansissel** [comment]
@ernetas thanks for helping reiterate and support my sanity here :)

**Comment by zabbal** [comment]
https://lintian.debian.org/manual/index.html - this tool might help with packages QA to ensure that they are good enough for inclusion into official Debian and Ubuntu repositories which would greatly help to reduce the maintenance burden for both devs and users.

**Comment by cfergus** [comment]
For clarity, can you say which release is intended by the "next logstash-forwarder release". ie  0.4.0

**Comment by jordansissel** [comment]
@zabbal fpm's test suite already uses lintian to ensure that packages built reasonably respect lintian's policy complaints.

> for inclusion into official Debian and Ubuntu repositories

I have no desire to have logstash-forwarder available in Ubuntu or Debian's default repositories. I feel, and have experience that, this would increase burdens on both users and developers.

**Comment by jordansissel** [comment]
> For clarity, can you say which release is intended by the "next logstash-forwarder release". ie 0.4.0

I don't really understand the question, but if it helps, I will add a milestone to this ticket.

**Comment by nwp90** [comment]
@jordansissel - FWIW, IMO you're quite right re. inclusion in Debian/Ubuntu. That would only perhaps become worthwhile if/when other packages start wanting to depend on your packages. Also quite right to break repo rather than keep it present with significantly outdated software. That way we all get the chance to notice and fix.

**Comment by jordansissel** [comment]
version 0.4.0 is released and the packages are back online.

Announcement here: http://www.elasticsearch.org/blog/logstash-forwarder-0-4-0-released/

For apt-get users, put this in /etc/apt/sources.list (This works for me Ubuntu 12.04)
```
deb http://packages.elasticsearch.org/logstashforwarder/debian stable main
```

For yum users, point here:

```
http://packages.elasticsearch.org/logstashforwarder/centos
```

**Comment by andrewsomething** [comment]
Thanks so much for all your work on this @jordansissel!

**Comment by jbraithwaite** [comment]
Thanks @jordansissel :+1:

**Comment by proffalken** [comment]
Just as I'd fudged it in my ansible roles! :P 

Thanks for sorting this :)

**Comment by shanmer** [comment]
Thanks, just putting together a puppet manifest to test :)

**Comment by gtonic** [comment]
Thanks a lot!

**Comment by stevedwray** [comment]
1. Please add a 32 bit package in the repo.
2. There should be a source package in the repo so that apt-get source logstash-forwarder works (because in the absence of a 32 bit package I'd like to be able to build one).

**Comment by dav-m85** [comment]
+1 Well done !

**Comment by luhkevin** [comment]
`deb https://packages.elasticsearch.org/logstashforwarder/debian stable main`

is dead for me. 

`deb https://packages.elasticsearch.co/logstashforwarder/debian stable main`

is what works. Is this documented somewhere?

**Comment by jk0** [comment]
There used to be a GHI for it a while back, but I don't remember where it's
at offhand. They didn't originally intent to use the repo for production,
so there have been a few hiccups along the way.

**Comment by ruflin** [comment]
Side note here: There is a deb and rpm repo for filebeat: https://www.elastic.co/guide/en/beats/libbeat/current/setup-repositories.html
